### ISSUE TITLE
Create custom Emulator backend

### ISSUE THREAD

**Issue by uhbif19** [mlabs-haskell/cem-script#66]
As we see CLB is too slow for fuzzing #60 and it does not seem to be fixable.

Due to most of Tx correctness checks should be validated by our off-chain code anyway (blocked by #65), emulator can be implemented by only recording current UTxO state. That is: if we do not count Plutus fees, which would be out of scope of this task.

**TODO:**

 - [ ] Implement UTxO store and its querying
 - [ ] Add support for ADA selection and minada enforcement without  `makeTransactionBodyAutoBalance`
 - [ ] Support submitting
 - [ ] Support time
 - [ ] Parameterize tests around backends
